pkg/columns/filter/filter.go

```diff
@@ -143,7 +143,8 @@ func GetFilterFromString[T any](cols columns.ColumnMap[T], filter string) (*Filt
 		fs.value = filterRule
 	}
 
-	if fs.comparisonType == comparisonTypeRegex && column.Kind() != reflect.String {
+	if fs.comparisonType == comparisonTypeRegex &&
+		(column.Kind() != reflect.String && column.Kind() != reflect.Array && column.Kind() != reflect.Slice) {
 		return nil, fmt.Errorf("tried to apply regular expression on non-string column %q", fs.column.Name)
 	}
```

Comment:
I know the logic was there before, but can't we lift this limitation and allow using regexp with non-string columns too? In that case, we'll need convert the field to string, using `GetFieldAsString`, before matching the regexp.

Reply:
Done. Although I am thinking if we should still have checks for `structs` ?

Reply:
Do you mean in case there is a struct within a struct in the event type? I think it makes sense, however I'll ping @flyth as he has the ideas more clear on this module.